docs/src/account/components.md

```diff
@@ -9,180 +9,201 @@ Account components are reusable units of functionality that define a part of an
-  whose contents must be provided at instantiation time through [`InitStorageData`](#initializing-placeholder-values).
-  If `values` are present, the entry is interpreted as a static map regardless of the `type` field, so
-  specifying `type = "map"` becomes purely descriptive in that case.
+`key-type` / `value-type` accept either a string type identifier (e.g. `"word"`) or a 4-element array of felt schema descriptors.
 
-In the example, the third storage entry defines a static storage map with two initial entries, while
-the fourth entry (`procedure_thresholds`) is a templated map whose contents are supplied at
-instantiation time.
+If `default-values` is omitted, the map is populated at instantiation via [`InitStorageData`](#providing-init-values). When `default-values` are present, they act as defaults: init data can optionally add entries and override existing keys.
 
-##### Multi-slot value
+In the example, the third storage entry defines a static map and the fourth entry (`procedure_thresholds`) is populated at instantiation.
+
+##### Typed map example
 
-Multi-slot values are composite values that exceed the size of a single slot (i.e., more than one `word`).
+You can type maps at the slot level via `key-type` and `value-type` (each a `WordSchema`):
 
-For multi-slot values, the following fields are expected:
+```toml
+[[storage.map]]
+name = "demo::typed_map"
+key-type = "word"
+value-type = "auth::rpo_falcon512::pub_key"
+```
 
-- `slots`: Specifies the list of contiguous slots that the value comprises
-- `values`: Contains the initial storage value for the specified slots
+This declares that all keys are `word` and all values are `auth::rpo_falcon512::pub_key`, regardless of whether the map contents come from `default-values = [...]` (static) or are supplied at instantiation via `InitStorageData`.
 
-Placeholders can currently not be defined for multi-slot values. In our example, the fifth entry defines a two-slot value.
+`key-type` / `value-type` are validated when building map entries from `InitStorageData` (and when validating `default-values`).
 
-#### Initializing placeholder values
+##### Multi-slot value
 
-When a storage entry introduces placeholders, an implementation must provide their concrete values
-at instantiation time. This is done through `InitStorageData` (available as `miden_objects::account::InitStorageData`), which can be created programmatically or loaded from TOML using `InitStorageData::from_toml()`.
+Multi-slot values are currently unsupported by component schemas.
 
-For example, the templated map entry above can be populated from TOML as follows:
+#### Providing init values
+
+When a storage entry requires init-supplied values, an implementation must provide their concrete values
+at instantiation time. This is done through `InitStorageData` (available as `miden_objects::account::component::InitStorageData`), which can be created programmatically or loaded from TOML using `InitStorageData::from_toml()`.
```

Comment:
Probably not for this PR, and I know we've been using these naming for a while - but I don't really love `InitStorageData` or "init data requirements". I don't really have great alternatives though. We could potentially use "Bootstrap" - e.g. `BootstrapData`, "bootstrap requirements" - but I'm not sure these are much better.